builtin/credential/userpass/path_users.go

```diff
@@ -43,15 +43,32 @@ func pathUsers(b *backend) *framework.Path {
 		Callbacks: map[logical.Operation]framework.OperationFunc{
 			logical.DeleteOperation: b.pathUserDelete,
 			logical.ReadOperation:   b.pathUserRead,
-			logical.UpdateOperation:  b.pathUserWrite,
+			logical.UpdateOperation: b.pathUserWrite,
+			logical.CreateOperation: b.pathUserWrite,
 		},
 
+		ExistenceCheck: b.userExistenceCheck,
+
 		HelpSynopsis:    pathUserHelpSyn,
 		HelpDescription: pathUserHelpDesc,
 	}
 }
 
-func (b *backend) User(s logical.Storage, n string) (*UserEntry, error) {
+func (b *backend) userExistenceCheck(req *logical.Request, data *framework.FieldData) (bool, error) {
+	username := data.Get("username").(string)
+	if username == "" {
+		return false, fmt.Errorf("missing username")
+	}
+
+	entry, err := req.Storage.Get("user/" + strings.ToLower(username))
```

Comment:
You should use the same logic here that you're using in `userCreateUpdate`, just to be safe in case something diverges. So that probably means using `b.user()` here.